src/main/java/org/highj/data/stateful/AsyncIO.java

```diff
@@ -0,0 +1,51 @@
+package org.highj.data.stateful;
+
+import org.derive4j.hkt.__;
+import org.derive4j.hkt.__2;
+import org.highj.data.Either;
+import org.highj.data.Maybe;
+import org.highj.data.stateful.async_io.*;
+import org.highj.data.tuple.T0;
+import org.highj.function.F1;
+
+public interface AsyncIO<E,A> extends __2<AsyncIO.µ,E,A> {
```

Comment:
Maybe you can add a short javadoc description how `AsyncIO` should be used, and what the restrictions are.